packages/framework/src/Form/OrderItemsType.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace Shopsys\FrameworkBundle\Form;
+
+use Shopsys\FrameworkBundle\Form\Admin\Order\OrderItemFormType;
+use Shopsys\FrameworkBundle\Form\Admin\Order\OrderPaymentFormType;
+use Shopsys\FrameworkBundle\Form\Admin\Order\OrderTransportFormType;
+use Shopsys\FrameworkBundle\Model\Order\Item\OrderItemPriceCalculation;
+use Shopsys\FrameworkBundle\Model\Order\Order;
+use Shopsys\FrameworkBundle\Model\Payment\PaymentFacade;
+use Shopsys\FrameworkBundle\Model\Transport\TransportFacade;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\CollectionType;
+use Symfony\Component\Form\Extension\Core\Type\FormType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class OrderItemsType extends AbstractType
+{
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Transport\TransportFacade
+     */
+    private $transportFacade;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Payment\PaymentFacade
+     */
+    private $paymentFacade;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Order\Item\OrderItemPriceCalculation
+     */
+    private $orderItemPriceCalculation;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Transport\TransportFacade $transportFacade
+     * @param \Shopsys\FrameworkBundle\Model\Payment\PaymentFacade $paymentFacade
+     * @param \Shopsys\FrameworkBundle\Model\Order\Item\OrderItemPriceCalculation $orderItemPriceCalculation
+     */
+    public function __construct(
+        TransportFacade $transportFacade,
+        PaymentFacade $paymentFacade,
+        OrderItemPriceCalculation $orderItemPriceCalculation
+    ) {
+        $this->transportFacade = $transportFacade;
+        $this->paymentFacade = $paymentFacade;
+        $this->orderItemPriceCalculation = $orderItemPriceCalculation;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $payments = $this->paymentFacade->getVisibleByDomainId($options['order']->getDomainId());
```

Comment:
I would extract the variable and annotate it for better PHPStan and IDE support:

```php
/** @var \Shopsys\FrameworkBundle\Model\Order\Order $order */
$order = $options['order'];

Reply:
done